diff --git a/lib/lz4.c b/lib/lz4.c
index 654bfdf..9182719 100644
--- lib/lz4.c
+++ lib/lz4.c
@@ -566,7 +566,7 @@ static unsigned LZ4_NbCommonBytes (reg_t val)
 * The ARM64EC ABI does not support AVX/AVX2/AVX512 instructions, nor their relevant intrinsics
 * including _tzcnt_u64. Therefore, we need to neuter the _tzcnt_u64 code path for ARM64EC.
 ****************************************************************************************************/
-#         if defined(__clang__) && (__clang_major__ < 10)
+#         if defined(__clang__)
             /* Avoid undefined clang-cl intrinsics issue.
              * See https://github.com/lz4/lz4/pull/1017 for details. */
             return (unsigned)__builtin_ia32_tzcnt_u64(val) >> 3;
