diff --git a/src/microui.c b/src/microui.c
index a8a3f74..631172d 100644
--- src/microui.c
+++ src/microui.c
@@ -697,12 +697,15 @@ void mu_update_control(mu_Context *ctx, mu_Id id, mu_Rect rect, int opt) {
   }
 }

-
 void mu_text(mu_Context *ctx, const char *text) {
+  mu_Color color = ctx->style->colors[MU_COLOR_TEXT];
+  mu_text_color(ctx, text, color);
+}
+
+void mu_text_color(mu_Context *ctx, const char *text, mu_Color color) {
   const char *start, *end, *p = text;
   int width = -1;
   mu_Font font = ctx->style->font;
-  mu_Color color = ctx->style->colors[MU_COLOR_TEXT];
   mu_layout_begin_column(ctx);
   mu_layout_row(ctx, 1, &width, ctx->text_height(font));
   do {
diff --git a/src/microui.h b/src/microui.h
index 78a1a34..446064e 100644
--- src/microui.h
+++ src/microui.h
@@ -275,6 +275,7 @@ void mu_update_control(mu_Context *ctx, mu_Id id, mu_Rect rect, int opt);
 #define mu_begin_panel(ctx, name)         mu_begin_panel_ex(ctx, name, 0)

 void mu_text(mu_Context *ctx, const char *text);
+void mu_text_color(mu_Context *ctx, const char *text, mu_Color color);
 void mu_label(mu_Context *ctx, const char *text);
 int mu_button_ex(mu_Context *ctx, const char *label, int icon, int opt);
 int mu_checkbox(mu_Context *ctx, const char *label, int *state);
