diff -u delabella/delabella.h delabella-patched/delabella.h
--- delabella.h
+++ delabella.h
@@ -6,6 +6,8 @@
 #ifndef DELABELLA_H
 #define DELABELLA_H
 
+#include <cstddef>
+
 template<typename T = double, typename I = int>
 struct IDelaBella2
 {
diff -u delabella/delabella.cpp delabella-patched/delabella.cpp
--- delabella.cpp
+++ delabella.cpp
@@ -26,1 +26,2 @@
 #include <string.h>
+#include <time.h>
